File: libs/shared/generate-ui-types/src/lib/generator-schema.ts
import { Option } from '@nx-console/shared/schema';
import { NxVersion } from '@nx-console/shared/nx-version';

export type GeneratorSchema = {
  collectionName: string;
  generatorName: string;
  description: string;
  options: Option[];
  context?: GeneratorContext;
};

export type GeneratorContext = {
  project?: string;
  normalizedDirectory?: string;
  directory?: string;
  prefillValues?: Record<string, string>;
  fixedFormValues?: Record<string, string>;
  nxVersion?: NxVersion;
};

File: libs/intellij/models/src/main/kotlin/dev/nx/console/models/GeneratorSchema.kt
package dev.nx.console.models

import kotlinx.serialization.Serializable

@Serializable
data class GeneratorSchema(
    val collectionName: String,
    val generatorName: String,
    val description: String,
    val options: List<NxGeneratorOption>,
    val context: NxGeneratorContext?
) {}
